LX/src/main/java/heronarts/lx/LXSnapshot.java

```diff
@@ -0,0 +1,243 @@
+package heronarts.lx;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import heronarts.lx.modulator.LXModulator;
+import heronarts.lx.parameter.CompoundParameter;
+import heronarts.lx.parameter.LXParameter;
+
+public class LXSnapshot extends LXComponent {
+
+  public Integer snapshotID;
+
+  public final List<LXSnapshotValue> snapshotValues = new ArrayList<LXSnapshotValue>();
```

Comment:
Note the pattern used in other LX classes for providing public visibility to a mutable list

  private final List<LXCompoundModulation> mutableModulations = new ArrayList<LXCompoundModulation>();
  public final List<LXCompoundModulation> modulations = Collections.unmodifiableList(this.mutableModulations);